doc/api/deprecations.md

```diff
@@ -2617,6 +2617,22 @@ async function openAndClose() {
 }
 ```
 
+<a id="DEP0XXX"></a>
+### DEP0XXX: `process.mainModule`
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: REPLACEME
+    description: Documentation-only deprecation.
+-->
+
+Type: Documentation-only
+
+[`process.mainModule`][] is a CJS-only feature while `process` global object is
+shared with non-CJS environment.
```

Comment:
```suggestion
[`process.mainModule`][] is a CommonJS-only feature while `process` global object is
shared with non-CJS environment. It's use within ESM modules is unsupported.
```

Reply:
I tend to avoid the phrasing “ESM modules”, I read the word “module“ twice in my head. Do you find the following good enough?

```suggestion
[`process.mainModule`][] is a CommonJS-only feature while `process` global object is
shared with non-CJS environment. It's use within ECMAScript modules is unsupported. 
```